core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -148,26 +163,14 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
                               std::is_convertible_v<IndexType2, member_type>),
                              bool> = false>
   inline RangePolicy(const IndexType1 work_begin, const IndexType2 work_end,
-                     Args... args)
+                     const ChunkSize chunk_size)
       : RangePolicy(typename traits::execution_space(), work_begin, work_end,
-                    args...) {}
-
- private:
-  inline void set() {}
+                    chunk_size) {}
 
  public:
-  template <class... Args>
-  inline void set(Args...) {
-    static_assert(
-        0 == sizeof...(Args),
-        "Kokkos::RangePolicy: unhandled constructor arguments encountered.");
-  }
-
-  template <class... Args>
-  inline void set(const ChunkSize& chunksize, Args... args) {
+  inline void set(ChunkSize chunksize) {
```

Comment:
Drop the function.  We have a `set_chunkl_size()` and don;t really need this one

Reply:
That is a breaking change, and `set(ChunkSize)` is a different interface than `set_chunk_size(int)`.

I don't mind making the change I just wanted to point out that it is reasonable for folks to call `set(ChunkSize)` (I have no idea if they actually do or not).

Reply:
We can deprecate `set()` but I wouldn't outright remove it.

Reply:
Yes we would need to go through the normal deprecation process.  It is ok to do as a follow up if you prefer.

Reply:
I just did it now.  It really is part of this.